internal/test/e2e/setup.go

```diff
@@ -136,6 +136,11 @@ func (e *E2ESession) setup(regex string) error {
 		return err
 	}
 
+	ipPool := e.ipPool.ToString()
+	if ipPool != "" {
```

Comment:
Might be an edge case but should we error here if the `ipPool` is empty?

Reply:
No because not all providers currently use ip pool. For instance, docker and cloudstack are not using it yet, so it's ok to be empty on those test cases.

Reply:
oh right, ignore this comment then. This should be good to merge